src/emotion-tracking/emotion-tracking.service.ts

```diff
@@ -0,0 +1,130 @@
+import { Injectable, HttpException, HttpStatus } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { EmotionalRecord } from 'src/entities/emotionalRecord.entity';
+import { EmotionHistory, Emotion } from './emotion-tracking.interface';
+import { EmotionTrackingStatusDto } from './dto/emotion-tracking.dto';
+import { User } from 'src/entities/user.entity';
+import { Module } from 'src/entities/module.entity';
+import moment from 'moment';
+import {UserService} from '../user/user.service'
+
+@Injectable()
+export class EmotionTrackingService {
+    constructor(
+        @InjectRepository(EmotionalRecord)
+        private emotionRecordRepository: Repository<EmotionalRecord>,
+        @InjectRepository(User)
+        private userRepository: Repository<User>,
+        @InjectRepository(Module)
+        private moduleRepository: Repository<Module>,
+        private userService: UserService
```

Comment:
Why do you need both user service and repository?

Reply:
because I want to get an elderly object with "modules" object inside it

Reply:
ขอโฆษณาหน่อย
`userService.findOne` works exactly like `userRepository.findOne`  with extra optional error handler.
Example usage: 
`this.userService.findOne({uid: uid}, {relations: ['modules']});` 

If you want the function to handle user not found: 
`this.userService.findOne({uid: uid}, {relations: ['modules'],shouldExist:true});` 

If you want the function to handle invalid user type
`this.userService.findOne({uid: uid}, {relations: ['modules'],shouldBeElderly:true});`

Reply:
thank youu